ui/src/main/webapp/app/gaffer/type-service.js

```diff
@@ -51,21 +51,39 @@ angular.module('app').factory('types', ['config', 'common', function(config, com
 
     var customShortValue = function(fields, parts) {
         var showWithLabel = (fields.length !== 1)
-
-        return fields.map(function(field) {
-            var layers = field.key.split('.');
-            var customValue = parts;
-            for (var i in layers) {
-                customValue = customValue[layers[i]];
-            }
-
-            customValue = service.getShortValue(customValue);
-
-            if (showWithLabel) {
-                return field.label + ': ' + customValue;
-            }
-            return customValue;
-        }).join(', ');
+        if (fields.length > 1) {
+            return fields.map(function(field) {
```

Comment:
In order to deduplicate the logic, It would be better to do something like:
```javascript
var showWithLabel = (fields.length > 1)
var parsedFields = fields.map(function(field) {
    // logic to create the field
});

if (showWithLabel) {
   return parsedFields.join(", ")
} else if (parsedFields.length == 1) {
   return parsedFields[0]
} else {
    // throw error if admin has configured custom object with no fields
}
```

Reply:
Thats a good idea, just pushed with those changes.